healpix_alchemy/tests/test_util.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from sqlalchemy.ext.declarative import declared_attr
+
+from ..util import InheritTableArgs
+
+
+@pytest.mark.parametrize('base_table_args,expected_table_args', [
+    [None, (1, {'bar': 2})],
+    [[], (1, {'bar': 2})],
+    [{}, (1, {'bar': 2})],
+    [[0], (0, 1, {'bar': 2})],
+    [[{'foo': 1}], (1, {'foo': 1, 'bar': 2})],
+    [[0, {'foo': 1}], (0, 1, {'foo': 1, 'bar': 2})],
+])
+def test_inherit_table_args(base_table_args, expected_table_args):
+
+    class Base:
+        if base_table_args is not None:
+            __table_args__ = base_table_args
+
+    class Derived(InheritTableArgs, Base):
+        @declared_attr
+        def __table_args__(cls):
+            *args, kwargs = super().__table_args__
```

Comment:
Why catch the lhs `args` with `*`?

Reply:
The SQLAlchemy `__table_args__` attribute can be any of the following:

1. a (possibly empty) tuple of table positional arguments
2. a (possibly empty) dictionary of table keyword arguments
3. a tuple, of which all but the last element (possibly numbering zero) are positional arguments, the last element of which is a (possibly empty) dictionary of keyword arguments

Option 3 is the most general. The purpose of this mixin is to normalize any of the above three styles to the last and most general style to simplify correctly implementing a model class or mixin that inherits from and extends its parent's `__table_args__`.

The statement

```python
*args, kwargs = super().__table_args__
```

normalizes the superclass' `__table_args__` to the third and most general format (that's the RHS of the assignment) and stores the resulting positional arguments in `args` and keyword arguments in `kwargs` so that you can extend them.